python/lsst/meas/algorithms/matcherSourceSelector.py

```diff
@@ -0,0 +1,171 @@
+#
+# LSST Data Management System
+# Copyright 2008-2015 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+
+from lsst.afw import table
+import lsst.pex.config as pexConfig
+from .sourceSelector import BaseSourceSelectorConfig, BaseSourceSelectorTask, sourceSelectorRegistry
+from lsst.pipe.base import Struct
+from functools import reduce
+
+
+class MatcherSourceSelectorConfig(BaseSourceSelectorConfig):
+    sourceFluxType = pexConfig.Field(
+        doc="Type of source flux; typically one of Ap or Psf",
+        dtype=str,
+        default="Ap",
+    )
+    minSnr = pexConfig.Field(
+        dtype=float,
+        doc="Minimum allowed signal-to-noise ratio for sources used for matching "
+        "(in the flux specified by sourceFluxType); <= 0 for no limit",
+        default=40,
+    )
+
+
+class MatcherSourceSelectorTask(BaseSourceSelectorTask):
+    """
+    !Select sources that are useful for astrometry.
+
+    Good astrometry sources have high signal/noise, are non-blended, and
+    did not have certain "bad" flags set during source extraction. They need not
+    be PSF sources, just have reliable centroids.
+    """
+    ConfigClass = MatcherSourceSelectorConfig
+
+    def __init__(self, *args, **kwargs):
+        BaseSourceSelectorTask.__init__(self, *args, **kwargs)
+
+    def selectSources(self, sourceCat, matches=None):
+        """
+        !Return a catalog of sources: a subset of sourceCat.
```

Comment:
Ok, I guess we'll hold off on reformatting as numpydoc (I did astrometrySS before that RFC had been fully settled). New python code is allowed (an I believe recommended) to use numpydoc instead of doxygen.